src/pages/constructor/constructor.js

```diff
@@ -0,0 +1,233 @@
+import React, { useEffect, useRef } from 'react';
+import { FormControl, InputLabel, NativeSelect } from '@material-ui/core';
+import mergeImages from 'merge-images';
+import { useDispatch, useSelector } from 'react-redux';
+import { useStyles } from './constructor.style';
+import { CONSTRUCTOR_TITLES } from '../../translations/constructor.translations';
+import {
+  getConstructorModelById,
+  getModelForConstructor,
+  setModelLoading
+} from '../../redux/constructor/constructor-model/constructor-model.actions';
+import { getConstructorBasic } from '../../redux/constructor/constructor-basic/constructor-basic.actions';
+import { getConstructorPattern } from '../../redux/constructor/constructor-pattern/constructor-pattern.actions';
+import { getConstructorBottom } from '../../redux/constructor/constructor-bottom/constructor-bottom.actions';
+import { getConstructorFrontPocket } from '../../redux/constructor/constructor-front-pocket/constructor-front-pocket.actions';
+import Loader from '../../components/loader';
+
+const { MODEL, BASIC, PATTERN, BOTTOM } = CONSTRUCTOR_TITLES[0];
+
+const Constructor = () => {
+  const styles = useStyles();
+  const dispatch = useDispatch();
+
+  const {
+    constructorModel,
+    constructorBasic,
+    constructorFrontPocket,
+    constructorPattern,
+    constructorBottom,
+    modelLoading
+  } = useSelector(({ Constructor }) => ({
+    constructorModel: Constructor.constructorModel,
+    constructorBasic: Constructor.constructorBasic,
+    constructorFrontPocket: Constructor.constructorFrontPocket,
+    constructorPattern: Constructor.constructorPattern,
+    constructorBottom: Constructor.constructorBottom,
+    modelLoading: Constructor.constructorModel.modelLoading
+  }));
+
+  const image = useRef(null);
+
+  const currentModel = constructorModel.constructorModel;
+  const models = constructorModel.modelsForConstructor;
+  const basics = constructorModel.constructorModel.constructorBasic;
+  const patterns = constructorModel.constructorModel.constructorPattern;
+  const bottoms = constructorModel.constructorModel.constructorBottom;
+
+  useEffect(() => {
+    dispatch(getModelForConstructor());
+  }, [dispatch]);
+
+  useEffect(() => {
+    console.log(modelLoading);
+  }, [modelLoading]);
+
+  useEffect(() => {
+    if (models) {
+      dispatch(getConstructorModelById(models[1]._id));
+    }
+  }, [models]);
+
+  useEffect(() => {
+    if (currentModel) {
+      dispatch(setModelLoading(true));
```

Comment:
I would say that I don't like to see all of these dispatches in the components itself, so I let's try to move it to hook
`useConstructor` and it will return all needed props like 
`constructorBasic &&
      constructorBottom &&
      constructorFrontPocket &&
      constructorPattern`

And this component will be responsible only for creating image and not dispatching. What do you think ?

Reply:
fixed